src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -340,16 +336,46 @@ public async Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccoun
             }
         }
 
-        public Task<IReadOnlyList<IAccount>> GetAccountsAsync(
+        public async Task<IReadOnlyList<IAccount>> GetAccountsAsync(
             string clientID,
             string redirectUri,
             AuthorityInfo authorityInfo,
             ICacheSessionManager cacheSessionManager,
             IInstanceDiscoveryManager instanceDiscoveryManager)
         {
-            // runtime does not yet support account discovery
+            if (!_wamOptions.ListWindowsWorkAndSchoolAccounts)
+            {
+                _logger.Info("[WamBroker] Returning no accounts due to configuration option.");
+                return Array.Empty<IAccount>();
+            }
+
+            string correlationId = Guid.NewGuid().ToString();
+            List<IAccount> accounts = new List<IAccount>();
+            var requestContext = cacheSessionManager.RequestContext;
+            var instanceMetadata = await instanceDiscoveryManager.GetMetadataEntryAsync(authorityInfo, requestContext, false).ConfigureAwait(false);
+
+            using (var core = new NativeInterop.Core())
+            using (var discoverAccountsResult = await core.DiscoverAccountsAsync(clientID, correlationId).ConfigureAwait(false))
+            {
+                if (discoverAccountsResult.IsSuccess)
+                {
+                    foreach (var acc in discoverAccountsResult.Accounts)
+                    {
+                        string environment = acc.Environment; 
+                        bool matches = instanceMetadata.Aliases.Any(alias => string.Equals(alias, environment, StringComparison.OrdinalIgnoreCase));
+                        
+                        if(matches)
+                            accounts.Add(WamAdapters.ConvertToMsalAccount(acc, clientID));
+                    }
+                }
+                else
+                {
+                    _logger.Info("[WamBroker] No accounts returned from WAM.");
```

Comment:
Log at error level and try to include as much details as possible from the `discoverAccountsResult`